### DISCUSSION TITLE
Unable to change selected node color.

### DISCUSSION THREAD

**Discussion by sheepdoggie** [freeplane/freeplane#530] [Help]
The map I am working on does not allow me to change the color of the selected node in the map by changing the preferences. I was told that this is because there was  a style selected for the nodes, but I never intended to select any style and the menu option "Format --> Apply Style --> No style" shows selected. How do I re-enable changing the selected node color?

**Comment by euu2021** [comment]
Disable `Preferences…->Defaults->Selection colors->Display selected nodes in bubbles`

**Reply by euu2021** [reply]
@dpolivaev 

Maybe, in order to make the preferences more clear, there could be radio buttons (instead of the single checkbox):

[] Use the Selected node Style
[] Use bubble with those properties

**Comment by euu2021** [comment]
Thinking about it, I noticed that, maybe, your problem is the oposite of what I thought at first. So, you need to enable `Preferences…->Defaults->Selection colors->Display selected nodes in bubbles`

Anyway, try to play with that option and see if it solves the problem.

**Reply by euu2021** [reply]
If you enable it: the selected node will have a bubble with those preferences bellow the option.

If you disable it: the selected node will be shown like the "Selected Node" style, which format you can change in the Edit Style menu. 

If I remember correctly, that's how it works.

**Reply by sheepdoggie** [reply]
I have tried Display selected nodes in bubbles. The selected node is in a bubble that is outlined in the selected color and the interior of the bubble is in the background (no color added). But when Display selected nodes in bubbles is deselected, then the selected node appears in the default color (outline and interior), no matter what color is selected in preferences. I should add, that it is only this map of the several I am working on, that shows this behavior. All the others behave as expected with respect to the selected node color preferences.

**Reply by euu2021** [reply]
If `Display selected nodes in bubbles` is disabled, than you need to go into `Edit styles` and format the style `Selected node` as you like. 

I don't have Freeplane here, so I can't get screenshots to explain it better.

**Comment by macmarrum** [comment]
When `Preferences…->Defaults->Selection colors->Display selected nodes in bubbles` is turned off, Freeplane uses the colors defined on a special style.

You can change the style of a selected node in `Format->Manage Styles->Edit styles`. There is a **Predefined style** called `Selected node`. Once you set colors there and confirm (`OK` at the bottom).

BTW, which version of Freplane do you use?

**Reply by sheepdoggie** [reply]
Freeplane version: 1.10.3. BTW my operating system (Linux Mint 20) uses Flatpak to manage versions of Freeplane, and upgrades do not occur in sequence. That is, I went from 1.8.11 to 1.10.3 without receiving 1.9.*.

**Reply by sheepdoggie** [reply]
The problem is that each map has a separate Predefined style, which cannot be deleted. Attempting to delete the Predefined style for a map results in an error message. Which means that the Preferences -> Default -> Selection Colors is useless and  meaningless, because in actuality the Selection Colors are determined by the Predefined style that is individual to each map. That is not only confusing but unnecessarily tedious. It is fine to have a Predefined style for a map when needed, but there should be the option not to have one so that global preferences apply.

**Reply by macmarrum** [reply]
> Which means that the Preferences -> Default -> Selection Colors is useless and meaningless

This option `Preferences…->Defaults->Selection colors->Standard selected node color (fallback)` is useful in case of old mindmaps, such that do not hold a definition of `Selected node`.

> the Selection Colors are determined by the Predefined style that is individual to each map.

Some of my mindmaps use dark colors, others bright colors. If I had a single selected-node color for both, it would be poorly readable in one of them.

> It is fine to have a Predefined style for a map when needed, but there should be the option not to have one so that global preferences apply.

It's a question of application design. So far the direction for Freeplane has been to go away from per-application-instance formatting properties to per-mindmap formatting properties.

Some of the older versions of Freeplane allow per-applicaton-instance formatting properties. Looking at [history_en.txt](https://www.freeplane.org/info/history/history_en.txt), I can see `Selected style` was introduced in 1.9.1. You might try version 1.8.12. I believe version 1.9.7 allowed to leave colors undefined. Maybe this will also work for you.

https://sourceforge.net/projects/freeplane/files/freeplane%20stable/archive/

**Reply by dpolivaev** [reply]
> The problem is that each map has a separate Predefined style, which cannot be deleted

It is not a bug but a feature (c)

It makes sure that all maps have the same look on all computers.
If you want to change or synchronize map styles you can use various methods:

* Format->Manage Styles->Replace map styles by…
* Format->Manage Styles->Copy map styles from…
*  "Follow style changes" checkbox shown on the above dialogs and in a map creation dialog
